mypy/suggestions.py

```diff
@@ -136,6 +137,47 @@ def get_return_types(typemap: Dict[Expression, Type], func: FuncDef) -> List[Typ
     return finder.return_types
 
 
+class ArgUseFinder(TraverserVisitor):
+    """Find all the types of arguments that each arg is passed to.
+
+    This is extremely simple minded but might be effective anyways.
+    """
+    def __init__(self, func: FuncDef, typemap: Dict[Expression, Type]) -> None:
+        self.typemap = typemap
+        self.arg_types = {
+            arg.variable: [] for arg in func.arguments
+        }  # type: Dict[SymbolNode, List[Type]]
+
+    def visit_func_def(self, o: FuncDef) -> None:
+        # Skip nested functions
+        pass
+
+    def visit_call_expr(self, o: CallExpr) -> None:
+        if not any(isinstance(e, RefExpr) and e.node in self.arg_types for e in o.args):
+            return
+
+        typ = get_proper_type(self.typemap.get(o.callee))
+        if not isinstance(typ, CallableType):
+            return
+
+        formal_to_actual = map_actuals_to_formals(
+            o.arg_kinds, o.arg_names, typ.arg_kinds, typ.arg_names,
+            lambda n: AnyType(TypeOfAny.special_form))
+
+        for i, args in enumerate(formal_to_actual):
+            for arg_idx in args:
+                arg = o.args[arg_idx]
+                if isinstance(arg, RefExpr) and arg.node in self.arg_types:
+                    self.arg_types[arg.node].append(typ.arg_types[i])
+
+
+def get_arg_uses(typemap: Dict[Expression, Type], func: FuncDef) -> List[List[Type]]:
+    """Find all the types of arguments that each arg is passed to."""
```

Comment:
Instead of repeating the class docstring, I would just add a simple example here:
```
For situation like:

    def foo(x: int) -> None: ...
    def bar(x: str) -> None: ...
    def test(x, y):
        foo(x)
        bar(y)

This will return [[int], [str]].
```